### ISSUE TITLE
Entering OTP fails

### ISSUE THREAD

**Issue by IrealiTY** [goaaats/FFXIVQuickLauncher#23]
v.1.4.1.0

When I enter my OTP in the prompt it stops on the 2nd to last number and crashes with  the following error:

```
---------------------------
Login failed
---------------------------
Logging in failed, check your login information or try again.

System.Exception: Could not log in to oauth.

   at XIVLauncher.XIVGame.OauthLogin(String username, String password, String otp)

   at XIVLauncher.XIVGame.Login(String username, String password, String otp)

   at XIVLauncher.MainWindow.StartGame()

StartGame L334
```

**Comment by goaaats** [comment]
I should've fixed that already - can you build and breakpoint the first line of XIVGame.Login, then check if the full OTP is provided? If not, I'll give you a build later to test

**Comment by IrealiTY** [comment]
Looks like, even though it gets "stuck" and unresponsive, entering the final number and pressing Enter still launches the game.

**Comment by IrealiTY** [comment]
> I should've fixed that already - can you build and breakpoint the first line of XIVGame.Login, then check if the full OTP is provided? If not, I'll give you a build later to test

sorry didnt see your comment, can do that.

**Comment by KazukiMutou** [comment]
Got the same error while my game wasn't updated. I had to use the official launcher first and update the game after the servers came back up. After that it worked right away. This does mean that maintenance queue is currently non-functional, though.

**Comment by IrealiTY** [comment]
small update - I can't make my own built due to time constraints

**Comment by goaaats** [comment]
I've tried this, and it seems to still work fine for me - maybe it was a temporary issue at SE

**Comment by Outfenneced** [comment]
I'm not seeing an error message, but I also have the issue with the 6th digit not showing up. If you hit enter after inputting the full number it works fine, though. Will look into it a bit more by building from source and will come back with results.

**Comment by Outfenneced** [comment]
In response to my comment there and what I believe is the issue: 
It's just that the display doesn't update to show the last number while it's doing the processing to launch the application. To me it just looks like if I press enter it actually performs the action because of this delay (~5 seconds). This appears to be because the WPF form is not doing data binding in a MVVM fashion and is instead using events in a more MVC manner that actually trigger before the display updates.

The wrong login info, I believe, is exactly as the error message describes. The main issue here seems to be the OTP number not updating before the communication with FFXIV's servers which makes it look like something is wrong when it's not.

Personally, I would recommend converting from events to MVVM data binding, but that's pretty large change to the structure of the application. 

Another option could be to spawn a background process which does all the communication and launching so the event "finishes" while doing the background processing and updates that value. You'd have to add some additional logic to make sure the OTP textbox can't change again and accidentally launch 2 FFXIV's, but that could just be a simple editable bool set to false. You'd also have to wait for the task to end before closing the application, but that could probably just be an `await` on the launch task basically the line before exiting the process. This change is likely a lot less work.

TLDR: It appears the backing logic is fine, just the display of the view is the issue.

**Comment by goaaats** [comment]
Yeah the UI code is terrible, I just wanted to get it done and rewrite it later - it's on my to-do list including making login asynchronous. As you said, it should still log you in fine, it just looks wrong.

**Comment by Outfenneced** [comment]
As a note, this has been "fixed" in my opinion since the update where you press ok/enter to submit the OTP

**Comment by goaaats** [comment]
Alright, thanks for the feedback!
